src/schema/query/users.query.ts

```diff
@@ -1,38 +1,133 @@
-import { GraphQLList, GraphQLError, GraphQLID } from 'graphql';
+import { GraphQLList, GraphQLError, GraphQLID, GraphQLInt } from 'graphql';
+import { GraphQLJSON } from 'graphql-type-json';
 import { User } from '@models';
-import { UserType } from '../types';
+import { UserConnectionType, encodeCursor, decodeCursor } from '../types';
 import { AppAbility } from '@security/defineUserAbility';
 import { Types } from 'mongoose';
 import { logger } from '@services/logger.service';
 import { graphQLAuthCheck } from '@schema/shared';
 import { Context } from '@server/apollo/context';
+import { accessibleBy } from '@casl/mongoose';
+import checkPageSize from '@utils/schema/errors/checkPageSize.util';
+import getFilter from '@utils/filter/getFilter';
+import { CompositeFilterDescriptor } from '@const/compositeFilter';
+import getSortOrder from '@utils/schema/resolvers/Query/getSortOrder';
+
+/** Default page size */
+const DEFAULT_FIRST = 10;
+
+/** Default filter fields */
+const FILTER_FIELDS: { name: string; type: string }[] = [
+  {
+    name: 'roles',
+    type: 'ObjectId',
+  },
+  {
+    name: 'name',
+    type: 'text',
+  },
+];
+
+/** Available sort fields */
+const SORT_FIELDS = [
+  {
+    name: 'createdAt',
+    cursorId: (node: any) => node.createdAt.getTime().toString(),
+    cursorFilter: (cursor: any, sortOrder: string) => {
+      const operator = sortOrder === 'asc' ? '$gt' : '$lt';
+      return {
+        createdAt: {
+          [operator]: decodeCursor(cursor),
+        },
+      };
+    },
+    sort: (sortOrder: string) => {
+      return {
+        createdAt: getSortOrder(sortOrder),
+      };
+    },
+  },
+];
 
 /** Arguments for the users query */
 type UsersArgs = {
   applications?: string[] | Types.ObjectId[];
+  first?: number;
+  filter?: CompositeFilterDescriptor;
+  afterCursor?: string;
 };
 
 /**
  * List back-office users if logged user has admin permission.
  * Throw GraphQL error if not logged or not authorized.
  */
 export default {
-  type: new GraphQLList(UserType),
+  type: UserConnectionType,
```

Comment:
@estelafs @RenzoPrats @Matthis-M-ReliefApps 
you all missed that this query is also used by a surveyjs question, so this is breaking the question 😅

Reply:
Oh damn, completely missed that indeed :open_mouth: will be more careful about where the queries are used next time